Automattic/wp-calypso
# Plans 2023: Typed Async props and other prop cleanup

Fixes https://github.com/Automattic/wp-calypso/issues/78572

## Proposed Changes

- Types the async props passed to `plan-features-2023-grid` form `plans-features-main` (gotta love these naming conventions)
- Removes several redundant/unused properties
- Several TODOs added to be picked up (by author/me) in other work

## Testing Instructions



* Confirm `/start/plans` and `/plans/[paid and free]` work as before. 
* Try a tailored flow e.g. `/setup/link-in-bio`
* Try `plans/[woo site]` as that imports `plan-features-2023-grid` directly
* Notices, plan prices, features, etc. show as normal

## Pre-merge Checklist
